internal/bloblang/query/functions.go

```diff
@@ -779,15 +779,20 @@ var _ = registerSimpleFunction(
 			`root.id = nanoid("abcde", 54)`,
 		),
 	),
-	func(_ FunctionContext) (interface{}, error) {
-		nid, err := gonanoid.New()
-		if err != nil {
-			panic(err)
-		}
-		return nid, nil
-	},
+	true,
+	nanoidFunction,
+	ExpectIntArg(1),
+	ExpectStringArg(0),
 )
 
+func nanoidFunction(args ...interface{}) (Function, error) {
```

Comment:
@Jeffail So I really just want to do the JS thing but I'm not sure if that's possible here with the type system or your conventions

```js
const x = (...args) => f(...args)
```

Reply:
Go doesn't allow us to do that. Since your new function requires between zero and two arguments, you'll want to call `ExpectBetweenNAndMArgs(0, 2)` followed by `ExpectStringArg(0)` and `ExpectIntArg(1)` like you have now. You'll also want to document and add a test for `nanoid(54)`, which corresponds to calling `gonanoid.New(54)` as described [here](https://github.com/matoous/go-nanoid/blob/b658bf464e31bd49c50b5173aaa41c7bfb2461d9/gonanoid.go#L81-L85). Then, you'll need to add a switch, something like this:
```
switch len(args) {
case 0:
	return gonanoid.New()
case 1:
	return gonanoid.New(args[0].(int))
case 2:
	return gonanoid.New(args[0].(string), args[1].(int))
}
```
Finally, instead of panicking when `err != nil`, you'll want to wrap the whole thing in a ClosureFunction like so: https://github.com/Jeffail/benthos/blob/53435661b1703e7100e8f07bcb37d70940792acd/internal/bloblang/query/functions.go#L649-L651

I haven't actually tested these suggestions, so apologies if I got something wrong.